### ISSUE TITLE
Specify time period

### ISSUE THREAD

**Issue by redbar0n** [aantipov/moiva#235]
It's not possible to get an accurate overview over the trend if one is not able to zoom out, time-wise (i.e. specify time period of the graphs).

**Github star histories:**

Moiva shows back until Jan 2020:
https://moiva.io/?npm=react+svelte+vue

But it should be able to get the GitHub star histories further back (not sure what API they use though..):
https://star-history.t9t.io/#facebook/react&sveltejs/svelte&vuejs/vue

**NPM trends:**

npmtrends.io has the "Downloads in past..." dropdown, allowing you to select, various time periods, like "1 month" or "5 years", or "All time", which is very useful, since a narrow time view doesn't really work to show a trend:

https://www.npmtrends.com/react-vs-svelte-vs-vue

**Comment by aantipov** [comment]
Hi @redbar0n, thanks for the suggestions.

**GitHub Stars** is a new chart that I added a couple days ago. It requires a lot of calls to GitHub API to fetch all the required data for popular libraries. For example, the link you provided won't work on a page reload (w/o cache), because the GitHub Token gets expired. If I implemented it the same way, the chart would always be broken.
So I decided to load data only for >2020-01 period and load data portion by portion (per user request) and use Permanent Cache which is not free. I'll see how things go, wait until Moiva has cached data for most popular libraries and how much space it requires. Maybe in the future I'll show data for earlier periods.

As of **NPM Trends** I don't find useful the dropdown and having options like "1 month" or "3 months". I always set it to "5 years" or "All time" to see the bigger picture. Moiva shows data since 2017-01 which I think should be more than enough to see the trends and compare libraries. Having the dropdown would be unnecessary complication of UI, I think.

The goal of Moiva is not to show all the historic data. Its goal is to provide data necessary to see trends and being able to compare libraries

I'm open to further discussions.
For now, I close the issue.
